libexec/pyenv-init

```diff
@@ -155,12 +170,20 @@ function print_path() {
       # (SSH provides /etc/ssh/sshrc and ~/.ssh/rc for that but no-one seems to use them for some reason).
       # This has caused an infinite `bashrc` execution loop for those people in the below nested Bash invocation (#2367).
       # --norc negates this behavior of such a customized Bash.
-      echo 'PATH="$(bash --norc -ec '\''IFS=:; paths=($PATH); '
+      echo 'pyenv_new_path="$(bash --norc -ec '\''IFS=:; paths=($PATH); '
       echo 'for i in ${!paths[@]}; do '
       echo 'if [[ ${paths[i]} == "'\'\'"${PYENV_ROOT}/shims"\'\''" ]]; then unset '\'\\\'\''paths[i]'\'\\\'\''; '
       echo 'fi; done; '
       echo 'echo "${paths[*]}"'\'')"'
-      echo 'export PATH="'"${PYENV_ROOT}"'/shims:${PATH}"'
+      # if no_push_path is unset, always update path.
+      # else, check for differences between the PATH variable and our proposed path var (pyenv_new_path)
+      # we check for differences by using bash string substitution to remove all chars but ":"
+      # and checking for equality -- i.e. ensuring old and new paths have same number of components.
+      # If they are equal, it means pyenv shims were NOT found, and therefore need to be added.
+      if [ -z "$no_push_path" -o "x${PATH//[^:]/}" = "x${pyenv_new_path//[^:]/}" ]; then
+        echo 'export PATH="'"${PYENV_ROOT}"'/shims:${pyenv_new_path}"'
```

Comment:
Same here. You can check `$PATH` for an entry with `[[ :$PATH: == *":<entry>:"* ]]` .

Reply:
I agree it's complicated.

I suppose I could do your suggested contains check earlier (if no_push_path is set).  I wasn't sure if we could use `[[` `]]` in this file (since I didn't see them elsewhere).